cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -104,7 +104,7 @@ public Boolean call() throws Exception {
         if (namespaces.size() == 0) {
           return false;
         }
-        if (namespaces.size() == 1 && NamespaceMeta.DEFAULT.equals(namespaces.get(0))) {
+        if (namespaces.contains(NamespaceMeta.DEFAULT)) {
           return true;
         }
         throw new IllegalStateException("Unexpected namespaces: " + namespaces);
```

Comment:
update this error message then, saying "default" namespace wasn't found. though that shouldn't happen.